store-api/src/main/java/com/stormeye/event/api/resource/DelegatorResource.java

```diff
@@ -0,0 +1,111 @@
+            description = "Obtains a page of validator rewards that are sortable by timestamp, blockHeight and eraId")
+    ResponseEntity<PageResponse<DelegatorReward>> getDelegatorRewards(@Parameter(description = "The public key of the delegator whose rewards are to be obtained")
+                                                                      @PathVariable(value = "publicKey") final String publicKey,
+                                                                      @Parameter(description = "The number of the page to obtain, starting from 1")
+                                                                      @RequestParam(value = "page", defaultValue = "1", required = false) final int page,
+                                                                      @Parameter(description = "The number of validator rewards to retrieved in a page, defaults to 10")
+                                                                      @RequestParam(value = "size", defaultValue = "10", required = false) final int size,
+                                                                      @Parameter(description = "The name of the field to sort on")
+                                                                      @RequestParam(value = "order_by", defaultValue = TIMESTAMP, required = false) final DelegatorRewardSortableFields orderBy,
+                                                                      @Parameter(description = "The direction of the sort")
+                                                                      @RequestParam(value = "order_direction", defaultValue = "DESC", required = false) final Sort.Direction orderDirection) throws NoSuchAlgorithmException {
+
+        logger.debug("getDelegatorRewards publicKey {}, page {}, size {}, orderBy {}, orderDirection {}",
+                publicKey,
+                page,
+                size,
+                orderBy,
+                orderDirection
+        );
+
+        var request = PageRequest.of(page - 1, size, PageUtils.getSort(orderBy, orderDirection));
+
+        return ResponseEntity.ok(new PageResponse<>(delegatorRewardRepository.findByPublicKey(
+                PublicKey.fromTaggedHexString(publicKey), request)
+        ));
+    }
+
+    @GetMapping(value = "/delegators/{publicKey}/total-rewards", produces = MediaType.APPLICATION_JSON_VALUE)
+    @Operation(tags = "rewards'", summary = "Obtains a page of validator rewards",
+            description = "Obtains a page of validator rewards that are sortable by timestamp, blockHeight and eraId")
+    ResponseEntity<Long> getTotalDelegatorRewards(@Parameter(description = "The public key of the validator whose rewards are to be obtained")
+                                                  @PathVariable(value = "publicKey") final String publicKey) throws NoSuchAlgorithmException {
+
+        logger.debug("getTotalValidatorRewards publicKey {}", publicKey);
```

Comment:
validate the public key

Reply:
Will be done in https://github.com/stormeye2000/cspr-events-store/issues/23